File: FE/shopapp-angular/src/app/responses/order/order-detail.response.ts
export interface OrderDetailResponse {
    id: number;
    order_id: number;
    product_id: number;
    product_name: string;
    product_image: string;
    quantity: number;
    price: number;
    total_money: number;
    color: string;
    created_at: number[];
    updated_at: number[];
}

File: BE/shopapp/src/main/java/com/project/shopapp/DTOs/responses/OrderDetailResponse.java
package com.project.shopapp.DTOs.responses;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.project.shopapp.models.Order;
import com.project.shopapp.models.Product;
import lombok.*;

import java.math.BigDecimal;

@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class OrderDetailResponse {
    private Long id;

    @JsonProperty("order_id")
    private Long orderId;

    @JsonProperty("product_id")
    private Long productId;

    @JsonProperty("product_name")
    private String productName;

    @JsonProperty("product_image")
    private String productImage;

    @JsonProperty("quantity")
    private int quantity;

    @JsonProperty("price")
    private BigDecimal price;

    @JsonProperty("total_money")
    private BigDecimal totalMoney;

    @JsonProperty("color")
    private String color;
}
